inspect/list_entities/src/index.ts

```diff
@@ -0,0 +1,116 @@
+import { Loci } from "molstar/lib/mol-model/loci";
+import { OrderedSet } from "molstar/lib/mol-data/int";
+
+
+async function init() {
+    // Create viewer
+    const plugin = await createRootViewer();
+    
+    // Download PDB
+    const fileData = await plugin.builders.data.download(
+        { url: "https://models.rcsb.org/1PTH.bcif", isBinary: true }
+    );
+
+    // Load PDB and create representation
+    const trajectory = await plugin.builders.structure.parseTrajectory(fileData, "mmcif");
+    const presetStateObjects = await plugin.builders.structure.hierarchy.applyPreset(trajectory, "default");
+
+    if (!presetStateObjects) {
+        throw new Error("Structure not loaded");
+    }
+
+    // Get Structure object from the structure stateObject selector.
+    // The Structure object contains properties and accessors to the underlying molecular data such as chains, residues, atoms, etc.
+    const struct = presetStateObjects.structure.data!;
+
+    // Create a QueryContext to be reused for all queries
+    // Limits the queries to only look at the structure
+    const ctx = new QueryContext(struct)
+
+    
+    // ==== Number of Waters ====
+    // Created a query to select all residues that are water
+    // but only select 1 atom per water (ensuring a Singleton selection)
+    const waterQuery = Queries.generators.atoms({
+        'entityTest': ctx => StructureProperties.entity.type(ctx.element) === 'water',
+        'atomTest': ctx => StructureProperties.atom.type_symbol(ctx.element) === 'O'
+    })
+    // Can cast as a Singleton selection since we are only selecting 1 atom per water
+    const waterSelection = waterQuery(ctx) as StructureSelection.Singletons;
+    const numWaters = waterSelection.structure.atomicResidueCount;
+
+
+    // ==== Covalent ligand names and residue code ====
+    // Create a query expression for all ligands connected to the protein
+    const covalentLigandExp = MS.struct.filter.isConnectedTo({
+        0: StructureSelectionQueries.ligand.expression,         // All ligands
+        target: StructureSelectionQueries.protein.expression,   // All protein atoms
+        'bond-test': true  // Only atoms covalently bound to the protein
+    })
+    // Query the atoms with the context to get a StructureSelection
+    const covLigQuery = StructureSelectionQuery('only-covalent-ligands', covalentLigandExp).query;
+    const covLigSelection = covLigQuery(ctx);
+    // Assume ligands in structure have >1 atoms.
+    // Therefore, the StructureSelection must be a Sequence
+    const covLigStructures = (covLigSelection as StructureSelection.Sequence).structures;
+    // Retrieve each ligand name and residue code
+    const covLigNames: string[] = [];
+    const covLigRes: string[] = [];
+    covLigStructures.forEach(s => s.units.map(u => {
+        // Create a location for the first element of the ligand
+        // to retrieve structure properties
+        const location = Location.create(s, u, u.elements[0])
```

Comment:
I suspect that for efficiency reasons, it might be better to create one Location outside the loop and use `Location.set(location,...` inside the loop.
Maybe something that can be mentioned in a comment rather than exemplified for code legibility)

Reply:
Ended up splitting it up as this is a common scheme done in the code so its best to showcase it here. I added a few comments to make it clear what is happening